src/lib/communication/DY.Lib.Communication.Core.Lemmas.fst

```diff
@@ -212,70 +210,71 @@ let receive_confidential_proof #invs #a tr higher_layer_preds comm_keys_ids rece
 
 val sign_message_proof:
   {|cinvs:crypto_invariants|} ->
-  #a:Type -> {| parseable_serializeable bytes a |} ->
+  #a:Type0 -> {|config:comm_layer_core_config a|} ->
   tr:trace ->
-  sender:principal -> receiver:principal -> payload:a -> pk_receiver:option bytes ->
+  sender:principal -> receiver:principal -> payload:either a bytes -> pk_receiver:option bytes ->
   sk_sender:bytes -> nonce:bytes ->
   Lemma
   (requires (
-    let payload_bytes = serialize a payload in
-    has_communication_layer_crypto_predicates /\
+    (Some? pk_receiver <==> Inr? payload) /\ 
+    (None? pk_receiver <==> Inl? payload) /\
```

Comment:
Is there a reason to not work with `a`-typed payloads in both the encrypted and unencrypted cases?